clients.yml

```diff
@@ -873,6 +873,16 @@ project/releng/scriptworker/v2/iscript/tb-prod:
   scopes:
     - assume:worker-type:scriptworker-prov-v1/tb-signing-mac-v1
     - queue:worker-id:signing-mac-v1/tb-mac-v3-signing*
+project/releng/scriptworker/v2/mac-signing/prod/firefoxci-gecko-signing-mac14m2-dep:
+  description: Dep signing worker pool for Firefox on macOS 14 (M2). Handles macOS-specific signing tasks.
+  scopes:
+    - assume:worker-type:scriptworker-prov-v1/gecko-signing-mac14m2-prod
+    - queue:worker-id:gecko-signing-mac14m2-prod/signing-mac14m2-prod*
```

Comment:
This looks like a `dev` or `dep` (the naming sucks) client being granted `prod` scopes, so not good.

I'm trying to figure out what the format *should* be based on other clients, and it's a bit of a mess. I think I see mistakes in some of those other clients as well. But as best I can tell, it should be something like:
```
project/releng/scriptworker/v2/mac-signing/dev/firefoxci-gecko-t:
  description: Dep signing worker pool for Firefox on macOS 14 (M2). Handles macOS-specific signing tasks.
  scopes:
    - assume:worker-type:scriptworker-prov-v1/gecko-depsigning-mac14m2
    - queue:worker-id:gecko-depsigning-mac14m2/depsigning-mac14m2*
```

I'm not really sure what the values in puppet are or how much leeway we have to change things here. This is just a first stab.